src/azure-cli/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -21,15 +21,28 @@
 from cryptography.hazmat.primitives.serialization import load_pem_private_key, Encoding, PublicFormat
 from cryptography.exceptions import UnsupportedAlgorithm
 
-
 from azure.cli.core import telemetry
 from azure.cli.core.profiles import ResourceType
+from azure.cli.core.util import sdk_no_wait
 
 from ._validators import _construct_vnet, secret_text_encoding_values
 
 logger = get_logger(__name__)
 
 
+def _azure_stack_wrapper(cmd, client, function_name, *args, **kwargs):
+    if cmd.cli_ctx.cloud.profile == 'latest':
+        function_name = 'begin_' + function_name
+        no_wait = False
+        if 'no_wait' in kwargs:
+            no_wait = kwargs.pop('no_wait')
+        return sdk_no_wait(no_wait, getattr(client, function_name), *args, **kwargs)
+
+    if 'no_wait' in kwargs:
+        kwargs.pop('no_wait')
```

Comment:
Can 
```
no_wait = False
if 'no_wait' in kwargs:
    no_wait = kwargs.pop('no_wait')
```
be called once and moved above
`if cmd.cli_ctx.cloud.profile == 'latest':`?

Reply:
Good idea.